### PR TITLE
feat(v2): Phase E — CLI layer (SIGTERM contract, run --flow, nodes/flows, graph/report/repro/migrate, unified exit codes)

### PR DESCRIPTION
**Pull request by sonoh5n**
## Summary

Phase E delivers the v2 CLI layer on top of the Phase D runner, in three audited sessions:

### E0 — SIGTERM interrupt contract (`0881ae5`)
- SIGTERM during `Runner.run` now flushes `RunReport(status=failed)` + `job.failed` + `run.completed(failed)` within the 10s SIGKILL grace.
- Handler is installed only for `Runner.run`'s scope and restored afterward; non-main-thread execution is unaffected.
- Error catalog: appended `3004 RunInterrupted` (append-only honored; `docs/rdetoolkit/error_catalog.md` regenerated).

### E1 — `run --flow` / `nodes` / `flows` (`11e5f83`)
- `rdetoolkit run --flow pkg.mod:callable` with exit codes **0/1/2/3** (success / failed / partial / usage error). Legacy `run TARGET` path untouched.
- `--validate-only` runs lifecycle steps 1–3 only — the flow is never resolved nor called.
- `nodes list|describe|lint` (+`--json`): lint emits **E2001** (duplicate id), **E2005** (unstable id), **E2006** (duplicate reserved annotation) with catalog remediation embedded; recommended-name deviations are advisory only (R1; E2002/E2004 stay retired).
- `flows list|describe` (+`--json`); `core/registry` gains 3 additive accessors.

### E2 — `graph` / `report show` / `repro` / `migrate check` (`e9b9499`)
- `graph`: mermaid/html/json rendered as a **Call Sequence** from `RunReport.iterations[*].node_calls` only — no edge inference, no lineage (ADR-022), pure Python.
- `report show`: validates `schema_version` (warns on unknown, never fails).
- `repro export → import → run`: round-trips to an identical directory tree; archive is stdlib `zipfile` only (call log + inputs + config, Design §10).
- `migrate check`: best-effort AST analysis of v1 patterns; always exit 0 for diagnostics.
- Cross-command exit-code contract pinned by tests: v2 inspection commands exit only 0/3 (never 1/2); Click parse errors stay 2.

### v1 impact (all sanctioned, zero-diff otherwise)
- `cli/app.py`: E1's `run()` extension (the one sanctioned exception) + two additive registration blocks. Every other v1 command body is byte-identical; `cli/validate.py`, `cli/__init__.py` untouched. `rdetoolkit --help` still lists init / gen-invoice / make-excelinvoice / archive unchanged.
- `errors.py` / `.pyi`: append-only (3004).

## Verification

All three sessions independently audited (VERIFIED) in local session records; every figure below reproduced by the auditor, not taken from the implementer:

```
.venv/bin/tox -e py312-module
== 2539 passed, 4 skipped, 70 deselected, 1363 warnings in 104.18s ==
Required test coverage of 80% reached. Total coverage: 88.11%
py312-ruff: All checks passed / py312-mypy: Success (139 source files)
```

- v1 tests (`tests/` root): untouched and GREEN.
- Negative guards: no ADR-020/022 vocabulary, no `_core` import from v2, no `plan`/`debug-node`, no v1-archive-internal imports.
- New tests: +106 (E0: 3, E1: 41, E2: 57 + config-validation 5) under `tests/v2/` only.

## Debts / notes for reviewers

1. `.codex/hooks.json` was committed in `0881ae5` and contains machine-specific absolute paths — candidate for removal/gitignore (tooling file, no runtime impact).
2. `RunAggregator.record_failure()` dead-code cleanup → Phase F boundary.
3. `migrate check`'s v1-handler-class detection is a documented best-effort AST heuristic (no formal v1 base class exists); revisit only if Phase F's migration guide formalizes the pattern.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

___

### **PR Type**
Tests, Enhancement, Bug fix

___

### **Description**
- Add comprehensive v2 CLI test suites for all major commands
  - Coverage for `run --flow`, `nodes`, `flows`, `graph`, `report`, `repro`, `migrate`
  - Exit code contract and error handling thoroughly tested

- Implement shared CLI module discovery helper for v2 commands

- Extend error catalog with `RunInterrupted` (SIGTERM) error code 3004

- Ensure robust handling of CLI usage errors and edge cases

___

### Diagram Walkthrough

```mermaid
flowchart LR
  A["Add v2 CLI test suites"] -- "tests/v2/cli/*, tests/v2/report/*" --> B["Test coverage for all CLI commands"]
  C["Add _v2_common.py module loader"] -- "src/rdetoolkit/cli/_v2_common.py" --> D["Shared CLI import logic"]
  E["Extend error catalog with 3004"] -- "src/rdetoolkit/errors.py" --> F["SIGTERM RunInterrupted error"]
  G["CLI error/exit code handling"] -- "tests, helpers" --> B
```

<details> <summary><h3> File Walkthrough</h3></summary>

<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Tests</strong></td><td><details><summary>9 files</summary><table>
<tr>
  <td><strong>test_cli_run.py</strong><dd><code>Add full test suite for `run --flow` CLI command</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/nims-mdpf/rdetoolkit/pull/513/files#diff-df2cef2bb536f6396b37032dcc56d7bde0b8c30e49aa34f973e74d9d84f73676">+330/-0</a>&nbsp; </td>

</tr>

<tr>
  <td><strong>test_cli_nodes.py</strong><dd><code>Add test suite for `nodes` CLI (list/describe/lint)</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/nims-mdpf/rdetoolkit/pull/513/files#diff-20e9074b6e32a651453312cc4e3ef0fc7aa17a861334682f0b2efedc83db18f8">+289/-0</a>&nbsp; </td>

</tr>

<tr>
  <td><strong>test_cli_flows.py</strong><dd><code>Add test suite for `flows` CLI (list/describe)</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/nims-mdpf/rdetoolkit/pull/513/files#diff-94a13cf91862697155711839d723e080888592c88169cc8ee8280826c2cfd5cf">+94/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>test_cli_graph.py</strong><dd><code>Add test suite for `graph` CLI (formats, errors)</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/nims-mdpf/rdetoolkit/pull/513/files#diff-8a7b010f8697d7b5f2b5c23ad84375b17eb6ea6072ee19e1074d27d6e851c66a">+207/-0</a>&nbsp; </td>

</tr>

<tr>
  <td><strong>test_cli_report.py</strong><dd><code>Add test suite for `report show` CLI command</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/nims-mdpf/rdetoolkit/pull/513/files#diff-9a8cf6a1de4e24b5e6be4fbd60b462f6cc5f407caa67b9c0552809ea9332d217">+228/-0</a>&nbsp; </td>

</tr>

<tr>
  <td><strong>test_cli_repro.py</strong><dd><code>Add test suite for `repro export/import` CLI commands</code>&nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/nims-mdpf/rdetoolkit/pull/513/files#diff-33ffe01c56a988e05427198e19e7360c3c0419da9dff1e466b8d7d37c5d0c59a">+360/-0</a>&nbsp; </td>

</tr>

<tr>
  <td><strong>test_cli_migrate.py</strong><dd><code>Add test suite for `migrate check` CLI command</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/nims-mdpf/rdetoolkit/pull/513/files#diff-9c1527200093e2a0f9fd28de3d701095b6c50c2556707ca6b6a32a0734a44994">+284/-0</a>&nbsp; </td>

</tr>

<tr>
  <td><strong>test_cli_exit_codes.py</strong><dd><code>Add cross-command CLI exit code contract tests</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/nims-mdpf/rdetoolkit/pull/513/files#diff-f321ede7153abb46430f5fae40f8bd651c798e0301838df8f2a15f3161d0218b">+274/-0</a>&nbsp; </td>

</tr>

<tr>
  <td><strong>test_graph_render.py</strong><dd><code>Add unit tests for graph rendering logic</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/nims-mdpf/rdetoolkit/pull/513/files#diff-911c10f5dffa22fef7dd0399e6a9c80bdf3c77438143f6b0b55b4102833ee50b">+251/-0</a>&nbsp; </td>

</tr>
</table></details></td></tr><tr><td><strong>Enhancement</strong></td><td><details><summary>1 files</summary><table>
<tr>
  <td><strong>_v2_common.py</strong><dd><code>Add shared module loader for v2 CLI commands</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/nims-mdpf/rdetoolkit/pull/513/files#diff-fdd0f55e0b0d4a0175b23232e3f883ae27a7d8de83d306cd092927604a36a1da">+22/-0</a>&nbsp; &nbsp; </td>

</tr>
</table></details></td></tr><tr><td><strong>Bug fix</strong></td><td><details><summary>1 files</summary><table>
<tr>
  <td><strong>errors.py</strong><dd><code>Add SIGTERM RunInterrupted error code 3004 to catalog</code>&nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/nims-mdpf/rdetoolkit/pull/513/files#diff-5e103e472efb744323aa5c1600372df58d7b7e4ede062858e94ba0d9af02ec00">+10/-0</a>&nbsp; &nbsp; </td>

</tr>
</table></details></td></tr><tr><td><strong>Additional files</strong></td><td><details><summary>31 files</summary><table>
<tr>
  <td><strong>hooks.json</strong></td>
  <td><a href="https://github.com/nims-mdpf/rdetoolkit/pull/513/files#diff-82c735c2c8b1f488b042175dc6cdb87b0d096a73e37ce0e856be9839eaf2d3bf">+0/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>error_catalog.md</strong></td>
  <td><a href="https://github.com/nims-mdpf/rdetoolkit/pull/513/files#diff-34324ef835d6bfbbd6fb20522d07b6f626c877a270434fc030ab641c9c37c9f2">+1/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>_v2_common.pyi</strong></td>
  <td><a href="https://github.com/nims-mdpf/rdetoolkit/pull/513/files#diff-2ebae3a4720645cb1cb5e971431cab224fd69b2a0cc13a7e10f3b593fda026bb">+3/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>app.py</strong></td>
  <td><a href="https://github.com/nims-mdpf/rdetoolkit/pull/513/files#diff-722d7be11d87ea41848e21bf77041320fbb112b1b124bf114386928e15ede925">+48/-2</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>app.pyi</strong></td>
  <td><a href="https://github.com/nims-mdpf/rdetoolkit/pull/513/files#diff-1c4ce6751bebc68f74b8e9e7f1f5240527799c360d67cdbf15d618be435006ba">+12/-1</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>flows_cmd.py</strong></td>
  <td><a href="https://github.com/nims-mdpf/rdetoolkit/pull/513/files#diff-0c36092886603a7e04b5f794da020f69bf51447fe1402c7d2d3647f5de9530ea">+59/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>flows_cmd.pyi</strong></td>
  <td><a href="https://github.com/nims-mdpf/rdetoolkit/pull/513/files#diff-d7c45732fd45bc2be44ab1d69b89b28bbf03a62834c46173f55d4e304333eec2">+6/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>graph_cmd.py</strong></td>
  <td><a href="https://github.com/nims-mdpf/rdetoolkit/pull/513/files#diff-217b567c8d8e1d9b4322e9028573e5c2f232aa99b0fe759cd5139e28012688c0">+43/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>graph_cmd.pyi</strong></td>
  <td><a href="https://github.com/nims-mdpf/rdetoolkit/pull/513/files#diff-6dd32bf25313f193dbb94240c959d90b07ba39795cdc553ecd3ce18a0cc7ce6f">+7/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>migrate_cmd.py</strong></td>
  <td><a href="https://github.com/nims-mdpf/rdetoolkit/pull/513/files#diff-fef9df7a3eeda8031ad25816008a9b8523ca02195d44a80ea2dffa77ac377aab">+85/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>migrate_cmd.pyi</strong></td>
  <td><a href="https://github.com/nims-mdpf/rdetoolkit/pull/513/files#diff-304cacab6d2943c99aada5fc6c2805b6ab27135b44da4565f577973f5bdec014">+13/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>nodes_cmd.py</strong></td>
  <td><a href="https://github.com/nims-mdpf/rdetoolkit/pull/513/files#diff-495b684e6f0129029d6a09edf40bd31dabb1ef65e9aaabe3b8332f60c594c696">+114/-0</a>&nbsp; </td>

</tr>

<tr>
  <td><strong>nodes_cmd.pyi</strong></td>
  <td><a href="https://github.com/nims-mdpf/rdetoolkit/pull/513/files#diff-8f1aa4f29a0d17a356a17269138a83df42d70fff772ade4ff7d06e5c5f890a6b">+12/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>report_cmd.py</strong></td>
  <td><a href="https://github.com/nims-mdpf/rdetoolkit/pull/513/files#diff-61ee18a630b38986464703f45bc88e8406630db50a46be7216d65e54e36dccde">+48/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>report_cmd.pyi</strong></td>
  <td><a href="https://github.com/nims-mdpf/rdetoolkit/pull/513/files#diff-0b3d4bc9cea5f17ca0b0353c71577bd71baa880f913b55e0b3311dde5285a8ef">+7/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>repro_cmd.py</strong></td>
  <td><a href="https://github.com/nims-mdpf/rdetoolkit/pull/513/files#diff-b661ae9f1d8f04813b6aa4c7e4e899c80058ad85dfabd9f83f77a3568ab73b47">+104/-0</a>&nbsp; </td>

</tr>

<tr>
  <td><strong>repro_cmd.pyi</strong></td>
  <td><a href="https://github.com/nims-mdpf/rdetoolkit/pull/513/files#diff-8231ef9814e37ed04692aae6a0e5b200ea288b40d5676a8d44a7ac583dfd9892">+8/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>run_cmd.py</strong></td>
  <td><a href="https://github.com/nims-mdpf/rdetoolkit/pull/513/files#diff-faf315980d4823ac931622325fe2615b91e9d4fcba247f7ff15c6d8a4d26dc27">+94/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>run_cmd.pyi</strong></td>
  <td><a href="https://github.com/nims-mdpf/rdetoolkit/pull/513/files#diff-6c9548d64faf571c3dc685fe86ab79b48d28cf3af729df2ec4dc14171573477b">+11/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>registry.py</strong></td>
  <td><a href="https://github.com/nims-mdpf/rdetoolkit/pull/513/files#diff-7e5ea29c773e6a8332774d14a6fe17ad5ae50b50e4b31b5866c2030b44081f70">+19/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>registry.pyi</strong></td>
  <td><a href="https://github.com/nims-mdpf/rdetoolkit/pull/513/files#diff-0604241b3353f6c99ce8d879e1c4c7b0c17f63629b0becd44ea45116e38a50d4">+3/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>graph_render.py</strong></td>
  <td><a href="https://github.com/nims-mdpf/rdetoolkit/pull/513/files#diff-47b66295f0d7aae93ee94e1daa5d0e69de50eb6c6dc90ac7cd4e05a23ebadb7f">+89/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>graph_render.pyi</strong></td>
  <td><a href="https://github.com/nims-mdpf/rdetoolkit/pull/513/files#diff-c7b721d9faeab6970be53f153887c0c957b9f5afb12e1b23c7046745c5448578">+3/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>lifecycle.py</strong></td>
  <td><a href="https://github.com/nims-mdpf/rdetoolkit/pull/513/files#diff-885d10815c8d4a43ca31282ed0ebf9d9ed50c07d1132bb8de807010cd5ae2a1d">+68/-35</a>&nbsp; </td>

</tr>

<tr>
  <td><strong>__init__.py</strong></td>
  <td><a href="https://github.com/nims-mdpf/rdetoolkit/pull/513/files#diff-704fba6708a9cfee316866275d64892d329c88b764caf4a091bff5804ad17aa8">+1/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>__init__.py</strong></td>
  <td><a href="https://github.com/nims-mdpf/rdetoolkit/pull/513/files#diff-d7cd47fd013459645f58b89471ef177d7db02b68873d1cec7ded1170620903fc">+9/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>run_flows.py</strong></td>
  <td><a href="https://github.com/nims-mdpf/rdetoolkit/pull/513/files#diff-eed1863687aad76d9c08900dfecca4525eaf5bb94f9c5ecfe50e203bd456cbac">+94/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>test_cli_module_discovery.py</strong></td>
  <td><a href="https://github.com/nims-mdpf/rdetoolkit/pull/513/files#diff-8d38f605638f645624c2c54b704dfad9c22363e81b2e6b398960bf3c0b907706">+142/-0</a>&nbsp; </td>

</tr>

<tr>
  <td><strong>test_cli_run_config_validation.py</strong></td>
  <td><a href="https://github.com/nims-mdpf/rdetoolkit/pull/513/files#diff-cb1fb74edb14ee7401c65f6f67183dfd838ebaa23e660b06201dea8f9c156885">+38/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>test_registry.py</strong></td>
  <td><a href="https://github.com/nims-mdpf/rdetoolkit/pull/513/files#diff-19b6d85dbec35d9733a4758b89de879620a8314561a8ab4212ed1bfa995a6cea">+142/-0</a>&nbsp; </td>

</tr>

<tr>
  <td><strong>test_sigterm_contract.py</strong></td>
  <td><a href="https://github.com/nims-mdpf/rdetoolkit/pull/513/files#diff-8180b06ccf2de05950e9d2f7ada7b3f924f3e5464a849e0cf8917a40f66ca8d9">+199/-0</a>&nbsp; </td>

</tr>
</table></details></td></tr></tr></tbody></table>

</details>

___

### CODE REVIEW

**Comment by sonoh5n** [comment]
## Review response — all 4 findings addressed in `262144a`

Thanks for the thorough review. Each finding was implemented RED-first with a regression test, then independently re-verified locally (full `tox -e py312-module`: **2554 passed / 4 skipped, coverage 88.15%**; ruff/mypy clean).

**[P1] SIGTERM handler window (`runner/lifecycle.py`)** — Fixed as suggested. `started`/`run_id`/`event_sink.open` are now initialized *before* the custom handler is installed, and `signal.signal(...)` is immediately followed by the covered inner `try` (no statement in between), so every instant the handler is active is covered by the failed-RunReport → `job.failed` → `run.completed(failed)` → sink-close path. Regression test: SIGTERM delivered from inside the run-id factory (under a prior no-op handler) previously escaped `Runner.run`; now the pre-existing handler receives it and the E0 contract suite stays green.

**[P1] Registry discovery for standalone processes (`nodes`/`flows`)** — Added a repeatable `--module <dotted.module>` option to `nodes list|describe|lint` and `flows list|describe` (shared loader in `cli/_v2_common.py`). Modules are imported before enumeration; import failures exit 3 with the exception text. We deliberately stopped short of entry-point/plugin discovery — that is Phase F scope (plugin discovery session) and will build on this same loader.

**[P2] Mermaid temporal links (`report/graph_render.py`)** — Agreed with your ADR-022 reading: the prohibition covers *inferred* dataflow/producer edges, not *recorded* temporal order. The renderer now emits `call_A -->|seq| call_B` between consecutive calls within each tile (never across tiles, never inferred). The acceptance test was strengthened accordingly: exact consecutive-pair `seq` edges required, unlabeled/cross-tile edges forbidden, dataflow vocabulary still banned.

**[P2] Empty directories in repro archives (`cli/repro_cmd.py`)** — `_write_tree` now always writes an explicit trailing-slash `ZipInfo` entry for each required input-directory root and for every empty descendant, so `repro import` reconstructs a runnable, tree-equivalent layout for zero-input/empty-invoice cases.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

File: src/rdetoolkit/runner/lifecycle.py [hunk 1]
```diff
@@ -74,42 +79,55 @@ def run(self, flow_fn: Callable[..., Any], **overrides: Any) -> RunReport:
...
+            try:
+                previous_sigterm = signal.getsignal(signal.SIGTERM)
+                signal.signal(signal.SIGTERM, _raise_run_interrupted)
+                sigterm_installed = True
+            except ValueError:
+                pass
+
+            self.run_id = self._run_id_factory()
+            self.event_sink.open(self.run_id)
+            started = time.time()
```
[the remark this answers is not in the document]
Comment (sonoh5n) [reply src/rdetoolkit/runner/lifecycle.py]: Resolved before this comment landed — it was anchored to `e9b9499`. Since `262144a`, `started`/`run_id`/`event_sink.open()` are all initialized *before* the custom handler is installed, and `signal.signal(...)` is immediately followed by the covered inner `try` (no statement in between), so every instant the handler is active flows through the failed-RunReport → `job.failed` → `run.completed(failed)` → sink-close path. A regression test (SIGTERM fired from inside the run-id factory under a prior no-op handler) pins this.

File: src/rdetoolkit/cli/repro_cmd.py [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+
+@app.command("import")
+def import_run(
+    archive_path: Annotated[Path, typer.Argument(metavar="<archive>")],
+    target: Annotated[Path, typer.Argument(metavar="<target_dir>")],
+) -> None:
+    """Reconstruct archived run inputs beneath a fresh target directory."""
+    if target.exists() and any(target.iterdir()):
+        _usage_error(f"Import target is not empty: {target}")
```
[the remark this answers is not in the document]
Comment (sonoh5n) [reply src/rdetoolkit/cli/repro_cmd.py:89]: Fixed in `365cdea`: `repro import` now checks `target.exists() and not target.is_dir()` first and exits 3 with "Import target is not a directory" — regression test added (previously leaked `NotADirectoryError` as exit 1).

File: .codex/hooks.json [hunk 3]
```diff
@@ -1 +1 @@
-{"hooks":{"Stop":[{"matcher":"","hooks":[{"type":"command","command":"'/home/nims-mg/.agents/skills/agmsg/scripts/check-inbox.sh' 'codex' '/home/nims-mg/github/rdetoolkit-worktrees/v2-phase-d'","commandWindows":"$b=$env:GIT_BASH; if (-not $b) { $b=$env:AGMSG_BASH }; if (-not $b) { $b='C:\\Program Files\\Git\\bin\\bash.exe' }; & $b -lc '''/home/nims-mg/.agents/skills/agmsg/scripts/check-inbox.sh'' ''codex'' ''/home/nims-mg/github/rdetoolkit-worktrees/v2-phase-d'''"}]}]}}
\ No newline at end of file
+{"hooks":{"Stop":[{"matcher":"","hooks":[{"type":"command","command":"'/home/nims-mg/.agents/skills/agmsg/scripts/check-inbox.sh' 'codex' '/home/nims-mg/github/rdetoolkit-worktrees/v2-phase-e'","commandWindows":"$b=$env:GIT_BASH; if (-not $b) { $b=$env:AGMSG_BASH }; if (-not $b) { $b='C:\\Program Files\\Git\\bin\\bash.exe' }; & $b -lc '''/home/nims-mg/.agents/skills/agmsg/scripts/check-inbox.sh'' ''codex'' ''/home/nims-mg/github/rdetoolkit-worktrees/v2-phase-e'''"}]}]}}
```
[the remark this answers is not in the document]
Comment (sonoh5n) [reply .codex/hooks.json]: Fixed in `365cdea`: `.codex/hooks.json` is removed from version control and `.codex/` is gitignored. The file is machine-generated agmsg tooling config and is regenerated locally; nothing in the repo references it.
